web/client/epics/dimension.js

```diff
@@ -52,9 +52,16 @@ module.exports = {
                     .switchMap( domains => {
                         const dimensions = domainsToDimensionsObject(domains, layer);
                         if (dimensions && dimensions.length > 0) {
+                            /**
+                             * updating the time object in state.layers (from describeDomains),
+                             * while maintaning other dimensions information in state.layer,
+                             * it also creates a list of dimensions (from descibeDomains) in state.dimensions.
+                             *  */
+                            const timeDimensionData = dimensions.filter(d => d.name === 'time').reduce(el => el);
```

Comment:
I think this is going to fail if filtered dimensions array is empty, I usually use lodash head in these cases:

const timeDimensionData = head(dimensions.filter(d => d.name === 'time'));

Reply:
this whole observer is triggered if **dimensions.length > 0**  ( line 54 )

Reply:
but if dimensions does not contain a time dimension, the filtered array will be empty, and reduce would crash

Reply:
@offtherailz I can fix this with the localization PR, or should I do it in a separate PR?

Reply:
just tried in a dev console of Chome
```
[].reduce(e => e)
VM19016:1 Uncaught TypeError: Reduce of empty array with no initial value
    at Array.reduce (<anonymous>)
    at <anonymous>:1:4
(anonymous) @ VM19016:1
[].reduce(e => e, {})
{}
```
:)